File: app/src/main/java/com/boardgamegeek/io/model/SearchResult.java
package com.boardgamegeek.io.model;

import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Path;
import org.simpleframework.xml.Root;

@Root(name = "item")
public class SearchResult {
	/**
	 * Includes boardgame, boardgameexpansion, videogame, rpg, rpgitem
	 */
	@Attribute private String type;
	@Attribute public int id;
	@Path("name") @Attribute(name = "value") public String name;
	@Path("name") @Attribute(name = "type") public String nameType;
	@Path("yearpublished") @Attribute(name = "value") public String yearPublished;
}

File: app/src/main/java/com/boardgamegeek/model/SearchResult.kt
package com.boardgamegeek.model

data class SearchResult(
    val id: Int,
    val name: String,
    val yearPublished: Int,
    val nameType: NameType = NameType.Unknown,
) {
    enum class NameType {
        Primary,
        Alternate,
        Unknown,
    }
}
